common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerValidator.java

```diff
@@ -223,6 +223,45 @@ && verifySignature(authenticator.packageName)) {
         return null;
     }
 
+    /**
+     * Determines which app is the broker based on having the work account registration in Account Manager.
+     *
+     * Known issue: When we're in an AccountManager callback (Especially on older Android devices, i.e. Android 10)
+     * Android For Work throws a SecurityException when we're calling AccountManager.getAuthenticatorTypes()
+     * i.e. E.g. Company Portal main process can call this freely, broker process can call this freely, but once running in AccountManager,
+     * on the work profile (user != 0), apparently sometimes it tries to get accounts from user 0 (personal profile) and fails.
+     *
+     * In such case, this method will return null.
+     *
+     * @return PackageName of the broker
+     */
+    @NonNull
+    // TODO: Consolidate getCurrentActiveBrokerPackageName and getValidActiveBrokerPackageName into one.
+    public String getValidActiveBrokerPackageName() throws ClientException {
+        final String methodTag = TAG + ":getValidActiveBrokerPackageName";
+
+        final int numberOfAuthenticators;
+        try {
+            final AuthenticatorDescription[] authenticators = AccountManager.get(mContext).getAuthenticatorTypes();
+            numberOfAuthenticators = authenticators.length;
+            Logger.info(methodTag, numberOfAuthenticators + " Authenticators registered.");
+            for (AuthenticatorDescription authenticator : authenticators) {
+                if (BROKER_ACCOUNT_TYPE.equals(authenticator.type)) {
+                    verifySignatureAndThrow(authenticator.packageName);
+                    return authenticator.packageName;
+                }
+            }
+        } catch (final Exception exception) {
+            final ClientException clientException = new ClientException(UNKNOWN_ERROR, exception.getMessage());
+            Logger.error(methodTag, exception.getMessage(), exception);
+            throw clientException;
+        }
+        final String errorMessage = "None of the " + numberOfAuthenticators + " authenticators, is type: " + BROKER_ACCOUNT_TYPE;
+        final ClientException clientException = new ClientException(UNKNOWN_ERROR, errorMessage);
```

Comment:
We can probably have a more specific error code here I guess that indicates something like BROKER_NOT_INSTALLED